realm/realm-library/src/main/java/io/realm/internal/SharedRealm.java

```diff
@@ -161,10 +176,12 @@ long getNativePtr() {
 
     public void beginTransaction() {
         nativeBeginTransaction(nativePtr);
+        invokeListenerIfSchemaChanged();
     }
 
     public void commitTransaction() {
         nativeCommitTransaction(nativePtr);
+        invokeListenerIfSchemaChanged();
```

Comment:
But `invokeListenerIfSchemaChanged()` in `refresh()` is still needed I think? It might be changed by other process.